### DISCUSSION TITLE
OidcAgentClient Token Expiration

### DISCUSSION THREAD

**Discussion by nn15296** [iTwin/itwinjs-core#633] [Q&A] [answered]
When using the OidcAgentClient to generate an access token used when creating a new context (AuthorizedBackendRequestContext), which is subsequently used to download a briefcase using BriefcaseManager, it seems that if downloading very large models which may take several hours, the token can expire prior to the completion of the download. When this happens, the download fails. Is there any way to mitigate this? It doesn't seem to be possible to generate a new token and context and then just continue the download - the download will always have to start again with a new instance of the BriefcaseManager.download() method which means that the issue just happens again with the new download taking ages and the expiration of the token.

Thanks

**Comment by pmconne** [comment, marked as answer]
Waiting hours to download a briefcase doesn't sound fun even if the token *doesn't* expire before the download completes. Roughly how large is the briefcase?
@ramanujam-raman may be able to help with the token expiration issue.

**Reply by nn15296** [reply]
The problem originally first surfaced with a ~20GB briefcase, but it very much depends on internet speed and can occasionally cause issues with briefcases which are far smaller.

**Reply by pmconne** [reply]
I wonder if someone like @eligijusm can tell us whether the briefcases are compressed for download.

**Reply by nn15296** [reply]
Any update on this? The unauthorised error seems to be cropping up every now and then on smaller iModels as well.

**Reply by pmconne** [reply]
Hi @nn15296,

@khanaffan is working on #365 to rectify this issue. In the meantime as a temporary workaround, consider installing [AzCopy](https://docs.microsoft.com/en-us/azure/storage/common/storage-use-azcopy-v10) and defining the environment variable `AZCOPY_BINARY_PATH` to point to the location of the AzCopy executable. This should significantly reduce the time required to download a briefcase.

Sorry for the temporary inconvenience.
